engine/include/cubos/engine/ui/canvas/draw_list.hpp

```diff
@@ -0,0 +1,144 @@
+/// @file
+/// @brief Class @ref cubos::engine::UIDrawList.
+/// @ingroup ui-canvas-plugin
+
+#pragma once
+#include <vector>
+
+#include <cubos/core/gl/render_device.hpp>
+
+#include <cubos/engine/api.hpp>
+
+namespace cubos::engine
+{
+    /// @brief Holds a collections of draw commands and their data.
+    /// @ingroup ui-canvas-plugin
```

Comment:
Since this part of the code is a bit confusing to people who are not familiar with it, we should explain it a lil more in depth.
I think you can explain the overall mechanism in the plugin page, and just a link to it here.
```suggestion
    /// @brief Holds a collections of draw commands and their data.
    /// @note See the @ref ui-canvas-plugin documentation page for more information.
    /// @ingroup ui-canvas-plugin
```